ksqldb-streams/src/main/java/io/confluent/ksql/execution/streams/SourceBuilder.java

```diff
@@ -239,10 +260,23 @@ private SourceBuilder() {
     );
 
     final String stateStoreName = tableChangeLogOpName(source.getProperties());
+
+    final PhysicalSchema physicalSchemaToMaterialize = getPhysicalSchemaWithKeyAndPseudoCols(source);
+
+    final Serde<GenericRow> valueSchemaToMaterialize = getValueSerde(
+        buildContext, source, physicalSchemaToMaterialize);
+
+    final Serde<Windowed<GenericKey>> keySerdeToMaterialize = buildContext.buildKeySerde(
+        source.getFormats().getKeyFormat(),
+        windowInfo,
+        physicalSchema,
+        source.getProperties().getQueryContext()
```

Comment:
I noticed this independently now as I was writing `buildMaterialized`. This was an oversight by me, but tests passed anyways.  I don't immediately see why this would work though, this seems kind of big and I'm not sure how the tests wouldn't catch it.